src/migrations/20220812205424-new-ar-topics.js

```diff
@@ -0,0 +1,179 @@
+          `,
+        { transaction },
+      );
+    });
+  },
+
+  async down(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const loggedUser = '0';
+      const sessionSig = __filename;
+      const auditDescriptor = 'RUN MIGRATIONS';
+      await queryInterface.sequelize.query(
+        `SELECT
+          set_config('audit.loggedUser', '${loggedUser}', TRUE) as "loggedUser",
+          set_config('audit.transactionId', NULL, TRUE) as "transactionId",
+          set_config('audit.sessionSig', '${sessionSig}', TRUE) as "sessionSig",
+          set_config('audit.auditDescriptor', '${auditDescriptor}', TRUE) as "auditDescriptor";`,
+        { transaction },
+      );
+
+      // Disable audit logging
+      await queryInterface.sequelize.query(
+        `
+          SELECT "ZAFSetTriggerState"(null, null, null, 'DISABLE');
+          `,
+        { transaction },
+      );
+
+      // Delete all the new topics
+      await queryInterface.sequelize.query(
+        `DELETE FROM "Topics" 
+        WHERE name IN (
+            'Child Screening and Assessment',
+            'Teaching / Caregiving Practices',
+            'Disabilities Services',
+            'Ongoing Monitoring Management System',
+            'Training and Professional Development'
+        ) 
+        AND "deletedAt" IS NULL;`,
+        { transaction },
+      );
+
+      // Revert the values in the topics array column of ActivityReports to their previous values.
+      await Promise.all(topicMapping.map(({ oldName, newName }) => queryInterface.sequelize.query(
+        `UPDATE "ActivityReports"
+        SET 
+            topics = array_replace(topics, '${newName}', '${oldName}') 
+        WHERE topics @> '{"${newName}"}';`,
+        { transaction },
+      )));
+
+      // "Un-delete" the recently deleted topics.
+      await Promise.all(topicMapping.map(({ oldName }) => queryInterface.sequelize.query(
+        `UPDATE "Topics" 
+        SET 
+            "mapsTo" = null,
+            "deletedAt" = null 
+        WHERE name = '${oldName}' 
+        AND "deletedAt" = (
+            SELECT max("deletedAt") 
+            FROM "Topics" 
+            WHERE name = '${oldName}');`,
+        { transaction },
+      )));
+
+      // Now, any deleted topics that share this name should map to this undeleted topic
+      await Promise.all(topicMapping.map(({ oldName }) => queryInterface.sequelize.query(
+        `UPDATE "Topics" t1 
+        SET 
+            "mapsTo" = t2.id 
+        FROM "Topics" t2 
+        WHERE t1.name = '${oldName}' 
+        AND t1."deletedAt" IS NOT NULL 
+        AND t2.name = '${oldName}' 
+        AND t2."deletedAt" is NULL;`,
+        { transaction },
+      )));
+
+      // Enable audit logging
+      await queryInterface.sequelize.query(
+        `
+          SELECT "ZAFSetTriggerState"(null, null, null, 'ENABLE');
+          `,
+        { transaction },
+      );
```

Comment:
This one too

Reply:
The above looks like it enables logging. Do you mean instead for me to remove the bit that disables logging?

e.g. the below?

https://github.com/HHS/Head-Start-TTADP/blob/ef3e04df52a2cffc3fd3ea8fb912f84d5534a7c2/src/migrations/20220812205424-new-ar-topics.js#L112-L118

Reply:
(you cleared this up for me in slack)

Done: 27d833626837cac70a21055f656c099716128eec